AIX: Adoptium script ./make_any_platform ... --build-variant openj9 jdk17u fails

As part of the process of preparing some new systems for Adoptium jenkins userid I found that there is an issue with one line of code.

Replication process:
- xlcv16 patched at FP009
- start script - this will fail with:
```
Scanning dependencies of target jsigjnitest
[ 12%] Building C object runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/main.c.o
[ 12%] Linking CXX shared library ../../libbcvrelationships.so
/home/jenkins/temurin-build/workspace/build/src/openj9/runtime/tests/jsig/main.c:61:8: error: duplicate 'volatile' declaration specifier [-Werror,-Wduplicate-decl-specifier]
static volatile sig_atomic_t handlerCalled = 0;
       ^
1 error generated.
Error while processing /home/jenkins/temurin-build/workspace/build/src/openj9/runtime/tests/jsig/main.c.
[ 13%] Building C object runtime/tests/gp/CMakeFiles/gptest.dir/__/__/copyright.c.o
gmake[6]: *** [runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/build.make:63: runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/main.c.o] Error 1
gmake[5]: *** [CMakeFiles/Makefile2:11540: runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/all] Error 2
gmake[5]: *** Waiting for unfinished jobs....
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_03.c.o
[ 13%] Built target bcvrelationships
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_04.c.o
[ 13%] Linking C shared library ../../libgptest.so
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_05.c.o
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_06.c.o
[ 13%] Built target gptest
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_07.c.o
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_08.c.o
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_09.c.o
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/args_10.c.o
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/cdefs.c.o
[ 13%] Building C object runtime/tests/jniarg/CMakeFiles/jniargtests.dir/__/__/copyright.c.o
[ 13%] Linking C shared library libjvm.so
[ 13%] Built target jvm_redirect
[ 13%] Linking C shared library ../../libjniargtests.so
[ 13%] Built target jniargtests
gmake[4]: *** [Makefile:130: all] Error 2
gmake[3]: *** [/home/jenkins/temurin-build/workspace/build/src/closed/OpenJ9.gmk:505: build-j9] Error 2
gmake[2]: *** [/home/jenkins/temurin-build/workspace/build/src/closed/custom/Main.gmk:50: j9vm-build] Error 2

ERROR: Build failed for targets 'product-images test-image debug-image' in configuration 'aix-ppc64-server-release' (exit code 2)

No indication of failed target found.
Hint: Try searching the build log for '] Error'.
Hint: See doc/building.html#troubleshooting for assistance.

gmake[1]: *** [/home/jenkins/temurin-build/workspace/build/src/make/Init.gmk:315: main] Error 2
gmake: *** [/home/jenkins/temurin-build/workspace/build/src/make/Init.gmk:186: product-images] Error 2
OpenJDK make failed, archiving make failed logs
Copying build.log to TemurinLogsArchive
Copying make-support/failure-logs to TemurinLogsArchive
```
- after script fails: cd workspace/build/src/; make # this is to verify error
```
[ 13%] Linking C shared library ../../libj9thrnumanatives29.so
/home/jenkins/temurin-build/workspace/build/src/openj9/runtime/tests/jsig/main.c:61:8: error: duplicate 'volatile' declaration specifier [-Werror,-Wduplicate-decl-specifier]
static volatile sig_atomic_t handlerCalled = 0;
       ^
1 error generated.
Error while processing /home/jenkins/temurin-build/workspace/build/src/openj9/runtime/tests/jsig/main.c.
gmake[6]: *** [runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/build.make:63: runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/main.c.o] Error 1
gmake[5]: *** [CMakeFiles/Makefile2:11540: runtime/tests/jsig/CMakeFiles/jsigjnitest.dir/all] Error 2
gmake[5]: *** Waiting for unfinished jobs....
...
[ 18%] Building C object runtime/util/CMakeFiles/j9util.dir/aix/sethwprefetch.c.o
[ 18%] Building C object runtime/util/CMakeFiles/j9util.dir/ut_srphashtable.c.o
[ 18%] Linking CXX static library ../lib/libj9util.a
[ 18%] Built target jvmti_test_src
[ 18%] Built target j9util
gmake[4]: *** [Makefile:130: all] Error 2
gmake[3]: *** [/home/jenkins/temurin-build/workspace/build/src/closed/OpenJ9.gmk:505: build-j9] Error 2
gmake[2]: *** [/home/jenkins/temurin-build/workspace/build/src/closed/custom/Main.gmk:50: j9vm-build] Error 2

ERROR: Build failed for target 'default (exploded-image)' in configuration 'aix-ppc64-server-release' (exit code 2)
```
- The error is easy to explain for AIX - the compiler forbids using the keyword `volatile` twice for a variable declaration. And AIX already declares (or embeds) _volatile_ as part of the typedef for `sig_atomic_t`
```
$ find /usr/include -name \*.h | xargs grep sig_atomic_t
/usr/include/sys/signal.h:typedef volatile int sig_atomic_t; /* accessable as an atomic entity (ANSI) */
```

The fix merely establishes that it is AIX - and uses a declaration without the keyword `volatile`

--- a/runtime/tests/jsig/main.c
+++ b/runtime/tests/jsig/main.c
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2016, 2018 IBM Corp. and others
+ * Copyright (c) 2016, 2022 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -58,7 +58,15 @@
 
 #define PRIMARY_CALLED 1
 #define SECONDARY_CALLED 2
+#if !defined(_AIX)
 static volatile sig_atomic_t handlerCalled = 0;
+#else /* !defined(_AIX) */
+/*
+ * In /usr/include/sys/signal.h the keyword 'volatile'
+ * is included in the typedef of sig_atomic_t
+ */
+static sig_atomic_t handlerCalled = 0;
+#endif /* !defined(_AIX) */
 
 typedef jint (JNICALL *CreateVM)(JavaVM**, JNIEnv**, void*);
 

